include/dem/particle_particle_contact_force.h

```diff
@@ -1087,6 +1087,164 @@ class ParticleParticleContactForce
         normal_unit_vector);
   }
 
+  /**
+   * @brief Carries out the calculation of the particle-particle DMT contact
+   * force and torques based on the updated values in contact_info
+   *
+   * @param contact_info A container that contains the required information for
+   * calculation of the contact force for a particle pair in contact
+   * @param tangential_relative_velocity Tangential relative velocity
+   * @param normal_relative_velocity_value Normal relative contact velocity
+   * @param normal_unit_vector Contact normal unit vector
+   * @param normal_overlap Contact normal overlap
+   * @param particle_one_properties Properties of particle one in contact
+   * @param particle_two_properties Properties of particle two in contact
+   * @param normal_force Contact normal force
+   * @param tangential_force Contact tangential force
+   * @param particle_one_tangential_torque Contact tangential torque on particle one
+   * @param particle_two_tangential_torque Contact tangential torque on particle two
+   * @param rolling_resistance_torque Contact rolling resistance torque
+   */
+  inline void
+  calculate_DMT_contact(particle_particle_contact_info<dim> &contact_info,
```

Comment:
I do not like that this is a separate model from hertz. I find it harder to maintain this way. Also you repeat a lot of the code here and maybe it is not necessary. I would choose one of the following options: make it as a parameter for hertz (I would choose this one), or split hertz and call the first part here (which could be useful for other models, actually).

Reply:
I'm not sure I agree with this.
I understand that a large fraction of the 100 lines are repeated from the models, but I do not like the idea of making a switch with the hertz model, because then it makes the hertz model grow even more with multiple switches. Furthermore cohesion plays a role in the whole Coulomb limit (it changes the Coulomb limit) so it's not like it's a small term at one spot, it changes the entire logic.

I would rather us keep the seperate function for now instead of having a larger one with multiple ifs.
If we want to refactor it, we should do that in a next PR and refactor it the way it was done with the rolling friction (where the rolling friction is added on top of the model and not mingled with the model).
First though, we would need to refactor the particle-wall contacts which are still a mess, but this is also for another PR.

Reply:
Perfect

Reply:
I thought about it and I agree with Bruno. Yeah p-w contacts files need definitively a refactor to fit with the p-p contacts architecture. The p-p forces would need some clean up, tho. The equations are long and difficult to read and there's ways to make it cleaner. However I think Gabo did a great work here that fits with the implementation of other forces and the refactoring should be in other PRs